drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -393,7 +396,22 @@ static int nrf5_tx(const struct device *dev,
 		nrf_802154_transmit_csma_ca_raw(nrf5_radio->tx_psdu);
 		break;
 	case IEEE802154_TX_MODE_TXTIME:
-	case IEEE802154_TX_MODE_TXTIME_CCA:
+	case IEEE802154_TX_MODE_TXTIME_CCA: {
+		bool cca = (mode == IEEE802154_TX_MODE_TXTIME_CCA);
+		uint32_t tx_at;
+
+		__ASSERT_NO_MSG(pkt);
+		__ASSERT(IS_ENABLED(CONFIG_NET_PKT_TIMESTAMP),
```

Comment:
Would be preferable to have a nrf dedicated Kconfig option, which would depend on CONFIG_NET_PKT_TIMESTAMP and be true by default. No need to assert here.

And I'd use if (IS_ENABLED(CONFIG_...)) { ... the logic you implemented here ...}, which itself could be in an inline function for better clarity.

Reply:
I added a dedicated Kconfig option `IEEE802154_NRF5_PKT_TXTIME` that depends on `NET_PKT_TXTIME` as discussed  under the comment https://github.com/zephyrproject-rtos/zephyr/pull/29123#discussion_r503263801
I also moved the logic to a separate function. I used preprocessor `#if ` instead of regular `if` so that the flow enters default clause when the symbol is undefined.